asserts/model.go

```diff
@@ -657,6 +694,119 @@ func checkOptionalSystemUserAuthority(headers map[string]interface{}, brandID st
 	return nil, fmt.Errorf("%q header must be '*' or a list of account ids", name)
 }
 
+func checkModelValidationSetAccountID(headers map[string]interface{}, brandID string) (string, error) {
+	accountID, err := checkOptionalStringWhat(headers, "account-id", "of validation-set")
+	if err != nil {
+		return "", err
+	}
+
+	// default to brand ID if account ID is not provided
+	if accountID == "" {
+		return brandID, nil
+	}
+	return accountID, nil
+}
+
+// checkModelValidationSetSequence reads the optional 'sequence' member, if
+// not set, returns 0 as this means unpinned. Unfortunately we are not able
+// to reuse `checkSequence` as it operates inside different parameters.
+func checkModelValidationSetSequence(headers map[string]interface{}, what string) (int, error) {
+	seq, err := checkIntWithDefaultWhat(headers, "sequence", what, 0)
+	if err != nil {
+		return 0, err
+	}
+
+	// Assert that sequence number is not zero, as zero will be interpreted as not specified,
+	// which means that the validation-set is unpinned.
+	if seq < 0 {
+		return 0, fmt.Errorf("\"sequence\" %s must be larger than 0", what)
+	}
+	return seq, nil
+}
+
+func checkModelValidationSetMode(headers map[string]interface{}, what string) (ModelValidationSetMode, error) {
+	modeStr, err := checkNotEmptyStringWhat(headers, "mode", what)
+	if err != nil {
+		return "", err
+	}
+
+	if modeStr != "" && !strutil.ListContains(validModelValidationSetModes, modeStr) {
+		return "", fmt.Errorf("\"mode\" %s must be %s, not %q", what, strings.Join(validModelValidationSetModes, "|"), modeStr)
+	}
+	return ModelValidationSetMode(modeStr), nil
+}
+
+func checkModelValidationSet(headers map[string]interface{}, brandID string) (*ModelValidationSet, error) {
+	accountID, err := checkModelValidationSetAccountID(headers, brandID)
+	if err != nil {
+		return nil, err
+	}
+
+	name, err := checkStringMatchesWhat(headers, "name", "of validation-set", validValidationSetName)
+	if err != nil {
+		return nil, err
+	}
+
+	what := fmt.Sprintf("of validation-set \"%s/%s\"", accountID, name)
+	seq, err := checkModelValidationSetSequence(headers, what)
+	if err != nil {
+		return nil, err
+	}
+
+	mode, err := checkModelValidationSetMode(headers, what)
+	if err != nil {
+		return nil, err
+	}
+
+	return &ModelValidationSet{
+		AccountID: accountID,
+		Name:      name,
+		Sequence:  seq,
+		Mode:      mode,
+	}, nil
+}
+
+func checkOptionalModelValidationSets(headers map[string]interface{}, brandID string) ([]*ModelValidationSet, error) {
+	valSets, ok := headers["validation-sets"]
+	if !ok {
+		return nil, nil
+	}
+
+	entries, ok := valSets.([]interface{})
+	if !ok {
+		return nil, fmt.Errorf(`"validation-sets" must be a list of validation sets`)
+	}
+
+	valSetName := func(vs *ModelValidationSet) string {
+		if vs.AccountID != "" {
+			return fmt.Sprintf("%s/%s", vs.AccountID, vs.Name)
+		}
+		return vs.Name
```

Comment:
because how things are implemented we can't reach here, no?

Reply:
Nice catch, that's true. I've simplified it and removed the lambda